lib/javascript/tools.js

```diff
@@ -227,120 +227,105 @@ function showEditDialog(edit_type, edit_id, edit_form_id, edit_title, refresh_ro
     parent.editType = edit_type;
     parent.editId = edit_id;
 
-    // Convert choices string ("tag0,tag1,tag2,...") to choices array
-    parent.editTagChoices = new Array();
-    if (tag_choices == undefined && tag_choices != '') {
-        // Load tag map
-        $.ajax(jsAjaxServer + '/ajax.server.php?page=tag&action=get_tag_map', {
-            success: function(data) {
-                tag_choices = $(data).find('content').text();
-                if (tag_choices != '') {
-                    showEditDialog(edit_type, edit_id, edit_form_id, edit_title, refresh_row_prefix);
-                }
-            }, type: 'post', dataType: 'xml'
-        });
-        return;
-    }
-	parent.editLabelChoices = new Array();
-    if (label_choices == undefined && label_choices != '') {
-        // Load tag map
-        $.ajax(jsAjaxServer + '/ajax.server.php?page=tag&action=get_labels', {
-            success: function(data) {
-                label_choices = $(data).find('content').text();
-                if (label_choices != '') {
-                    showEditDialog(edit_type, edit_id, edit_form_id, edit_title, refresh_row_prefix);
-                }
-            }, type: 'post', dataType: 'xml'
-        });
-        return;
-    }
-    var splitted = tag_choices.split(',');
-    var i;
-    for (i = 0; i < splitted.length; ++i) {
-        parent.editTagChoices.push($.trim(splitted[i]));
-    }
-	splitted = label_choices.split(',');
-	for (i = 0; i < splitted.length; ++i) {
-        parent.editLabelChoices.push($.trim(splitted[i]));
-    }
-
-    parent.dialog_buttons = {};
-    this.dialog_buttons[jsSaveTitle] = function () {
-        $.ajax({
-            url: parent.saveUrl,
-            type: 'POST',
-            data: $(parent.editFormId).serializeArray(),
-            success: function (resp) {
-                $('#editdialog').dialog('close');
+    $.when($.ajax(jsAjaxServer + '/ajax.server.php?page=tag&action=get_tag_map'), $.ajax(jsAjaxServer + '/ajax.server.php?page=tag&action=get_labels')).then(function( a1, a2 ) {
 
-                if (parent.refreshRowPrefix != '') {
-                    var new_id = $.trim(resp.lastChild.textContent);
+        if(a1[2].status != 200 || a2[2].status != 200){
+            //Some sort of error handling?
+            alert("Error");
+        }
+        parent.editTagChoices = [];
+        parent.editLabelChoices = [];
+        tag_choices = $(a1[0]).find('content').text();
+        label_choices = $(a2[0]).find('content').text();
+
+        var splitted = tag_choices.split(',');
+        var i;
+        for (i = 0; i < splitted.length; ++i) {
```

Comment:
This could be rewritten using an `Array.map` (`$.each` in jQuery if I remember correctly) and defining `parent.editTagChoices` to be `tag_choices`. This is a detail, not sure it is better.